pkg/vsphere/disk/disk_manager.go

```diff
@@ -302,8 +310,14 @@ func (m *Manager) Attach(op trace.Operation, disk *types.VirtualDisk) error {
 }
 
 // Detach attempts to detach a virtual disk
-func (m *Manager) Detach(op trace.Operation, d *VirtualDisk) error {
-	defer trace.End(trace.Begin(d.DevicePath))
+func (m *Manager) Detach(op trace.Operation, config *VirtualDiskConfig) error {
+	defer trace.End(trace.Begin(""))
+
+	d, err := NewVirtualDisk(config, m.Disks)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
```

Comment:
My comment is here because it's as close as I could get it to the line in question (#355 in the new code). I'm wondering why successfully decrementing the ref count is an error.